frontend/src/metabase/ui/components/inputs/SegmentedControl/SegmentedControl.styled.tsx

```diff
@@ -11,15 +11,53 @@ export const getSegmentedControlOverrides =
         size: "md",
         radius: rem(4),
       },
-      styles: (theme, { fullWidth }: SegmentedControlStylesParams) => {
+      styles: (
+        theme,
+        { fullWidth, shouldAnimate }: SegmentedControlStylesParams,
+      ) => {
         return {
           label: {
-            padding: fullWidth ? `${rem(8)} ${rem(16)}` : rem(8),
+            color: theme.fn.themeColor("text-dark"),
+            padding: fullWidth
+              ? `${theme.spacing.sm} ${theme.spacing.md}`
+              : theme.spacing.sm,
+            fontSize: theme.fontSizes.md,
             fontWeight: "normal",
             lineHeight: "1rem",
-            "&[data-active]": {
+            "&:hover": {
```

Comment:
Label color is the same on hover. Is this intentional? If so, then this block is redundant.